include/rosidl_dynamic_typesupport/api/dynamic_data.h

```diff
@@ -109,21 +143,29 @@ rosidl_dynamic_typesupport_dynamic_data_get_name(
 // DYNAMIC DATA CONSTRUCTION =======================================================================
 ROSIDL_DYNAMIC_TYPESUPPORT_PUBLIC
 rcutils_ret_t
-rosidl_dynamic_typesupport_dynamic_data_create_from_dynamic_type_builder(
+rosidl_dynamic_typesupport_dynamic_data_init_from_dynamic_type_builder(
   rosidl_dynamic_typesupport_dynamic_type_builder_t * dynamic_type_builder,
-  rosidl_dynamic_typesupport_dynamic_data_t ** dynamic_data);  // OUT
+  rcutils_allocator_t * allocator,
+  rosidl_dynamic_typesupport_dynamic_data_t * dynamic_data);  // OUT
 
 ROSIDL_DYNAMIC_TYPESUPPORT_PUBLIC
 rcutils_ret_t
-rosidl_dynamic_typesupport_dynamic_data_create_from_dynamic_type(
+rosidl_dynamic_typesupport_dynamic_data_init_from_dynamic_type(
   rosidl_dynamic_typesupport_dynamic_type_t * dynamic_type,
-  rosidl_dynamic_typesupport_dynamic_data_t ** dynamic_data);  // OUT
+  rcutils_allocator_t * allocator,
+  rosidl_dynamic_typesupport_dynamic_data_t * dynamic_data);  // OUT
 
 ROSIDL_DYNAMIC_TYPESUPPORT_PUBLIC
 rcutils_ret_t
 rosidl_dynamic_typesupport_dynamic_data_clone(
   const rosidl_dynamic_typesupport_dynamic_data_t * other_dynamic_data,
-  rosidl_dynamic_typesupport_dynamic_data_t ** dynamic_data);  // OUT
+  rcutils_allocator_t * allocator,
+  rosidl_dynamic_typesupport_dynamic_data_t * dynamic_data);  // OUT
+
+ROSIDL_DYNAMIC_TYPESUPPORT_PUBLIC
+rcutils_ret_t
+rosidl_dynamic_typesupport_dynamic_data_fini(
+  rosidl_dynamic_typesupport_dynamic_data_t * dynamic_data);
```

Comment:
For other reviewers, the vast majority of changes for this pr can be summed up with this diff, basically adding allocators where we do allocation and avoiding double pointers as out parameters (matching how most of the `rcl`-like interfaces work, and the rest is mostly mechanical changes related to this shift.